javatests/arcs/core/analysis/EntityTypeUnionTest.kt

```diff
@@ -0,0 +1,102 @@
+package arcs.core.analysis
+
+import arcs.core.data.EntityType
+import arcs.core.data.FieldType
+import arcs.core.data.Schema
+import arcs.core.data.SchemaFields
+import arcs.core.data.SchemaName
+import com.google.common.truth.Truth.assertThat
+import org.junit.Assert.assertTrue
+import org.junit.Assert.assertNotNull
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+@RunWith(JUnit4::class)
+class EntityTypeUnionTest {
+    // TODO(bgogul): hash for all Schema instances.
+
+    @Test
+    fun schemaUnionMergesNames() {
+        val emptySchemaFields = SchemaFields(emptyMap(), emptyMap())
```

Comment:
I have a feeling we'll need this type a lot. Might be worth defining it centrally somewhere so that we don't have to rewrite every creation of it if we change the representation.

e.g. something like `SchemaFields.getTopSchema()`

Reply:
I agree, we might need something like this. Will add it in a separate PR as the need arises. 

BTW, why `TopSchema`? Do you mean `Top` in the sense of the top value in an abstract domain?